pycloudlib/azure/cloud.py

```diff
@@ -683,46 +790,67 @@ def launch(
             # Check if we already have an existing network interface that is
             # not attached to a virtual machine. If we have, we will just
             # use it
-            nic = self._check_for_network_interfaces()
+            if not network_interface_clients:
+                nic = self._check_for_network_interfaces()
+
+        created_nics = [nic]
+        created_ip_addresses = []
 
         if nic is None:
+            created_nics.clear()
```

Comment:
Nit: Why not:

```diff
diff --git a/pycloudlib/azure/cloud.py b/pycloudlib/azure/cloud.py
index 889465b..41be201 100644
--- a/pycloudlib/azure/cloud.py
+++ b/pycloudlib/azure/cloud.py
@@ -786,18 +786,18 @@ class Azure(BaseCloud):
         # it to enable more ports. The rationale for is that we want to reuse
         # those resources only if they are generic enough
         nic = None
+        created_nics = []
         if not inbound_ports:
             # Check if we already have an existing network interface that is
             # not attached to a virtual machine. If we have, we will just
             # use it
             if not network_interface_clients:
                 nic = self._check_for_network_interfaces()
+                created_nics.append(nic)
 
-        created_nics = [nic]
         created_ip_addresses = []
 
         if nic is None:
-            created_nics.clear()
             self._log.debug(
                 "Could not find a network interface. Creating one now"
             )
```

It feels more natural to not have invalid values on that list, so that the code is more easy to read / extend.